client/src/components/DataFiles/DataFilesModals/DataFilesCompressModal.js

```diff
@@ -0,0 +1,197 @@
+  const history = useHistory();
+  const location = useLocation();
+  const dispatch = useDispatch();
+  const status = useSelector(
+    state => state.files.operationStatus.compress,
+    shallowEqual
+  );
+
+  const params = useSelector(
+    state => state.files.params.FilesListing,
+    shallowEqual
+  );
+
+  const isOpen = useSelector(state => state.files.modals.compress);
+  const selectedFiles = useSelector(
+    ({ files: { selected, listing } }) =>
+      selected.FilesListing.map(i => ({
+        ...listing.FilesListing[i]
+      })),
+    shallowEqual
+  );
+  const selected = useMemo(() => selectedFiles, [isOpen]);
+  const formRef = React.useRef();
+  const toggle = () =>
+    dispatch({
+      type: 'DATA_FILES_TOGGLE_MODAL',
+      payload: { operation: 'compress', props: {} }
+    });
+
+  const onOpened = () => {
+    dispatch({
+      type: 'FETCH_FILES_MODAL',
+      payload: { ...params, section: 'modal' }
+    });
+  };
+
+  const onClosed = () => {
+    dispatch({ type: 'DATA_FILES_MODAL_CLOSE' });
+    if (isString(status)) {
+      dispatch({
+        type: 'DATA_FILES_SET_OPERATION_STATUS',
+        payload: { status: {}, operation: 'compress' }
+      });
+      history.push(location.pathname);
+    }
+  };
+
+  const compressCallback = () => {
+    const { filename } = formRef.current.values;
+    dispatch({
+      type: 'DATA_FILES_COMPRESS',
+      payload: { filename, files: selected }
+    });
+  };
+
+  let buttonIcon;
+  if (status === 'RUNNING') {
+    buttonIcon = <LoadingSpinner placement="inline" />;
+  } else if (status === 'ERROR') {
+    buttonIcon = <Icon name="alert" />;
+  } else {
+    buttonIcon = null;
+  }
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onOpened={onOpened}
+      onClosed={onClosed}
+      toggle={toggle}
+      className="dataFilesModal"
+    >
+      <ModalHeader toggle={toggle}>Compress Files</ModalHeader>
+      <ModalBody>
+        <DataFilesCompressForm
+          formRef={formRef}
+          first={(selectedFiles[0] && selectedFiles[0].name) || ''}
+        />
+        <p>
+          A job to compress your files will be submitted on your behalf. You can
+          check the status of this job on your Dashboard, and your compressed
+          file will appear in this directory.
+        </p>
+        {status === 'SUCCESS' && (
+          <span style={{ color: 'green' }}>
+            Successfully started compress job
+          </span>
+        )}
+      </ModalBody>
+      <ModalFooter>
+        <Button
+          onClick={compressCallback}
+          className="data-files-btn"
+          disabled={status === 'RUNNING'}
+          styleName="submit-button"
+        >
+          {buttonIcon}
+          <span styleName={buttonIcon ? 'with-icon' : ''}>Compress</span>
+        </Button>
+        <Button
+          color="secondary"
+          className="data-files-btn-cancel"
+          onClick={toggle}
+        >
+          Close
```

Comment:
I think these Close buttons are being removed from all modals. My understanding is that Design is wanting us to rely on just the X button.